hatrac/model/storage/amazons3.py

```diff
@@ -32,19 +35,31 @@ def __init__(self, config):
         self.bucket_name = config['s3_bucket']
 
         # TODO: do something better for authz than storing secret in our config data!
-        self.conn_config = config['s3_connection']
-
+        self.access_key = config['s3_access_key']
+        self.secret_key = config['s3_secret_key']
+        
         # TODO: encode better identifiers if we ever support multiple
         # buckets or credentials and need separate sub-pools
         self.config_tuple = (self.bucket_name,)
+        
+        self.conn_tuple = None
 
     def _new_connection(self):
         """Open a new S3 connection and get bucket handle."""
-        conn = boto.S3Connection(**self.conn_config)
+        conn = S3Connection(self.access_key, self.secret_key)
         bucket = conn.get_bucket(self.bucket_name)
         return (conn, bucket)
-
-def s3_bucket_wrap(deferrered_conn_reuse=False):
+    
+    def _get_pooled_connection(self):
```

Comment:
Actually, my intention was to extend webauthn2.util.PooledConnection here but I forgot to specify the superclass and call its __init__ method above (which will take the self.config_tuple as argument).  Can you do that instead of implementing the _get_pooled_connection and _put_pooled_connection methods here?